### PR TITLE
Allow custom top bar widget to be "not always visible"

### PR DESCRIPTION
**Pull request by ulusoyca**
## Description
Currently, the custom top bar widget operates exclusively when the `isTopBarLayerAlwaysVisible` is set to `true`. Setting this to `false` when custom top bar is used triggers an exception. This PR introduces the flexibility to enable the custom top bar to be used even when `isTopBarLayerAlwaysVisible` is set to `true`.

### Changes Introduced:

* When a `WoltModalSheetPage.topBar` widget is provided, it will be wrapped with shadow and elevation determined by the theme or default settings. Without a `WoltModalSheetPage.topBar,` a default top bar will be generated, styled according to the theme settings.
* For pages utilizing the default top bar, the title of the top bar will also be utilized. In contrast, pages with a custom top bar will omit the top bar title provided with `WoltModalSheetPage.topBarTitle`.
* An example has been added to the playground app to showcase the application of a custom top bar.
* Readme has been updated and enhanced in line with these changes.

| `isTopBarLayerAlwaysVisible` false  | `isTopBarLayerAlwaysVisible` true |
| ------------- | ------------- |
|  <video src="https://github.com/woltapp/wolt_modal_sheet/assets/13782003/a8cafa78-7976-4d4c-83d9-04081d4bd50f"> | <video src="https://github.com/woltapp/wolt_modal_sheet/assets/13782003/c1e3fc93-1b84-429d-81b8-a62a8d3a5edc"> |

### CODE REVIEW

File: lib/src/content/components/main_content/wolt_modal_sheet_top_bar.dart [hunk 1]
```diff
@@ -2,35 +2,70 @@ import 'package:flutter/material.dart';
...
         (theme.brightness == Brightness.light ? const Color(0xFFE4E4E4) : const Color(0xFF121212));
-    final topBarHeight = page.navBarHeight ??
-        themeData?.navBarHeight ??
-        defaultThemeData.navBarHeight;
+    final topBarHeight =
+        page.navBarHeight ?? themeData?.navBarHeight ?? defaultThemeData.navBarHeight;
     final backgroundColor =
         page.backgroundColor ?? themeData?.backgroundColor ?? defaultThemeData.backgroundColor;
     final surfaceTintColor = themeData?.surfaceTintColor ?? defaultThemeData.surfaceTintColor;
+
```
Comment (TahaTesser) [inline lib/src/content/components/main_content/wolt_modal_sheet_top_bar.dart:41]:
```suggestion

    // Create the default top bar widget.
    SizedBox topBarWidget = SizedBox(
      height: topBarHeight - elevation,
      width: double.infinity,
    );

    // If page.topBar is not null, then wrap it in the default top bar widget.
    if (page.topBar != null) {
      topBarWidget = SizedBox(
        height: topBarHeight - elevation,
        width: double.infinity,
        child: page.topBar,
      );
    }

```

File: lib/src/content/components/main_content/wolt_modal_sheet_top_bar.dart [hunk 2]
```diff
@@ -2,35 +2,70 @@ import 'package:flutter/material.dart';
...
+                type: MaterialType.canvas,
+                color: backgroundColor,
+                surfaceTintColor: surfaceTintColor,
+                shadowColor: shadowColor,
+                elevation: elevation,
+                child: SizedBox(
+                  height: topBarHeight - elevation,
+                  width: double.infinity,
+                ),
+              ),
```
Comment (TahaTesser) [inline lib/src/content/components/main_content/wolt_modal_sheet_top_bar.dart:68]:
```suggestion
        Material(
          type: MaterialType.canvas,
          color: backgroundColor,
          surfaceTintColor: surfaceTintColor,
          shadowColor: shadowColor,
          elevation: elevation,
          child: topBarWidget,
        ),
```

File: lib/src/modal_page/wolt_modal_sheet_page.dart [hunk 3]
```diff
@@ -166,8 +166,11 @@ class WoltModalSheetPage {
     this.trailingNavBarWidget,
     this.hasTopBarLayer,
     this.isTopBarLayerAlwaysVisible,
-    this.topBarWidget,
-  }) : assert((singleChildContent != null) == (sliverList == null));
+    this.topBar,
+  })  : assert((singleChildContent != null) == (sliverList == null),
```
Comment (TahaTesser) [inline lib/src/modal_page/wolt_modal_sheet_page.dart:170]:
```suggestion
  })  : assert(singleChildContent == null || sliverList == null,
```

File: playground/lib/home/pages/sheet_page_with_custom_top_bar.dart [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      ),
+    );
+  }
+}
+
+/// A custom top bar with a gradient background and a search bar.
+class _CustomTopBar extends StatelessWidget {
+  const _CustomTopBar({required this.onClosed, required this.onBackPressed});
+
+  static const _searchBarPadding = EdgeInsets.only(right: 64, top: 16, bottom: 16, left: 0);
```
Comment (TahaTesser) [inline playground/lib/home/pages/sheet_page_with_custom_top_bar.dart:52]:
```suggestion
  static const _searchBarPadding = EdgeInsetsDirectional.only(start: 64, top: 16, bottom: 16, end: 0);
```

**Review by TahaTesser — APPROVED** [review]
LGTM with some minor fixes

File: playground/lib/home/pages/sheet_page_with_custom_top_bar.dart (hunk 4 above)
Comment (ulusoyca) [reply playground/lib/home/pages/sheet_page_with_custom_top_bar.dart:52]: Thanks for the directional support!

**Pull request merged**